### ISSUE TITLE
Integrate wasm-utils into nitro-protocol

### ISSUE THREAD

**Issue by geoknee** [statechannels/statechannels#2621]
I propose that we tackle converting our entire stack to use this wasm stuff in a separate PR. That should

- make it much easier to avoid accidentally using some slower utils 
- share the speed advantages with other applications (e.g. browser wallet)
- cut down on `thisTypeToThatType` gymnastics

and my rationale for separate PR is that there are some questions:
- will there be a circular dependency relationship between `native-utils` and `nitro-protocol`?
- should `native-utils` actually just be part of `nitro-protocol`?
- will it seamlessly run in the browser?

_Originally posted by @geoknee in https://github.com/statechannels/statechannels/pull/2591#issuecomment-698328195_

**Comment by andrewgordstewart** [comment]
> should native-utils actually just be part of nitro-protocol?

I don't think so. There is real overhead in the `native-utils` package: you must have an appropriate version of rust installed, and when you run `yarn install` you will compile a crate, for which you probably want a beefy computer to execute quickly, etc...

**Comment by andrewgordstewart** [comment]
> will it seamlessly run in the browser?

I'd bet (small) money on it!
